sdk/storage/Azure.Storage.Common/src/TaskExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Extensions to execute async Tasks synchronously.
+    /// </summary>
+    internal static class TaskExtensions
+    {
+        /// <summary>
+        /// Ensure the Task has finished executing.
+        /// </summary>
+        /// <typeparam name="T">Task's return type.</typeparam>
+        /// <param name="task">The task.</param>
+        /// <returns>The result of executing the task.</returns>
+        public static T EnsureCompleted<T>(this Task<T> task)
+        {
+            Debug.Assert(task.IsCompleted);
+            return task.GetAwaiter().GetResult();
```

Comment:
Don't we need a ConfigureAwait(false) here and below?

Reply:
I've fleshed this out to be more specific about when we're doing proper sync-using-Tasks and when we're _(temporarily until we have a sync transport layer)_ doing sync-over-async _(and should use a ConfigureAwait)_.

Reply:
We don't need ConfigureAwait() here the task is supposed to be completed at this point

Reply:
After thinking about it more, let's keep `ConfigureAwait` in case we missed some code path in testing and accidentally end up doing sync-over-async here.

Reply:
After talking with you this morning, I'm going to make the changes you suggest.  Thanks Pavel!